sdl2-vga-terminal/src/TuiTerminal.hpp

```diff
@@ -0,0 +1,66 @@
+/*************************************************************
+ ***           Text User Interface utilities               ***
+ *** ===================================================== ***
+ *** Some basic TUI components                             ***
+ *** if growing, to be promoted as a stand-alone library   ***
+ *** or could be just a component of this library          ***
+ *** ----------------------------------------------------- ***
+ *** Description:                                          ***
+ *** using VgaTerminal in mode3                            ***
+ *** - no other method allowed                             ***
+ *** - no access to VgaTerminal                            ***
+ *** - no access to Window                                 ***
+ *** - working with default palette colors only            ***
+ *** - Header display a title, description and TUI Version ***
+ *** - background                                          *** 
+ *** - fix Footer                                          ***
+ *** - no access to Window or VgaTerminal                  ***
+ *** ===================================================== ***
+ ***                  VERSION 0.1-showcase                 ***
+ *************************************************************/
+#pragma once
+
+#include "VgaTerminal.hpp"
+
+class TuiTerminal
+{
+public:
+    static constexpr char* VERSION = "0.1";
+
+    TuiTerminal(const std::string& title, const std::string& description);
+    void draw();
+    void render();
+    VgaTerminal::terminalChar_t background_tc = { 177, 8, 9 };
+    VgaTerminal::terminalChar_t footer_tc = { 219, 15, 1};
+    static constexpr uint8_t defaultCol = 15;
+    static constexpr uint8_t defaultBgCol = 1;
+    // TODO use a label object (component interface) for the body
+    void drawSingleBorderDialog(const SDL_Rect& r, uint8_t col, uint8_t bgCol, const std::string& title, const std::string body);
+    void drawDialog(const SDL_Rect& r, uint8_t col, uint8_t bgCol, const std::string& header, const std::string body);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Function parameter 'body' should be passed by const reference.](https://app.codacy.com/manual/Raffaello/sdl2-vga-terminal/pullRequest?prid=5472420)